harbors/schema.py

```diff
@@ -45,18 +45,20 @@ class Meta:
     municipality = graphene.String()
     image_file = graphene.String()
     servicemap_id = graphene.String()
-    zip_code = graphene.String()
-    phone = graphene.String()
-    email = graphene.String()
-    www_url = graphene.String()
-    image_link = graphene.String()
-    mooring = graphene.Boolean()
-    electricity = graphene.Boolean()
-    water = graphene.Boolean()
-    waste_collection = graphene.Boolean()
-    gate = graphene.Boolean()
-    lighting = graphene.Boolean()
-    suitable_boat_types = graphene.List(BoatTypeType)
+    zip_code = graphene.String(required=True)
+    phone = graphene.String(required=True)
+    email = graphene.String(required=True)
+    www_url = graphene.String(required=True)
+    image_link = graphene.String(required=True)
+    mooring = graphene.Boolean(required=True)
+    electricity = graphene.Boolean(required=True)
+    water = graphene.Boolean(required=True)
+    waste_collection = graphene.Boolean(required=True)
+    gate = graphene.Boolean(required=True)
+    lighting = graphene.Boolean(required=True)
+    suitable_boat_types = graphene.NonNull(
+        graphene.List(graphene.NonNull(BoatTypeType))
+    )
```

Comment:
this says that there is guaranteed to be a suitable boat type, which is not really technically true, although _usually_ true in terms of business logic. just `suitable_boat_types = graphene.List(BoatTypeType, required=True)` should be enough

Reply:
It depends how we represent the "no suitable boat types", if that is represented as empty array then I think it's correct because it generates the following contract:
`suitableBoatTypes: BoatTypesBerthsQuery_harbors_edges_node_properties_suitableBoatTypes[];`

if I apply `suitable_boat_types = graphene.List(BoatTypeType, required=True)` we get contract:
`suitableBoatTypes: (BoatTypesBerthsQuery_harbors_edges_node_properties_suitableBoatTypes | null)[];`
So it means that the array it self can contain null values, which is not nice I think.

What do you think?

Reply:
We btw get the same result with:
`suitable_boat_types = graphene.List(graphene.NonNull(BoatTypeType), required=True)`
which actually looks better.

Reply:
looking at it from the generated GQL schema:
- your current code generates `suitableBoatTypes: [BoatTypeType!]!` which is just not true, as we can't guarantee that there will always be items in the list
- my suggestion generates: `suitableBoatTypes: [BoatTypeType]!` which means it could be empty list or list with null values, as GQL does not differentiate unfortunately

Reply:
yeah, no-need to wrap `graphene.List` in `NonNull`, since you can pass `required=True`

Reply:
ok, fair enough! 👍  will fix it.

Reply:
As per the [Graphene docs](https://docs.graphene-python.org/en/latest/types/list-and-nonnull/#nonnull)

```python
import graphene

class Character(graphene.ObjectType):
    name = graphene.NonNull(graphene.String)
```

The previous NonNull code snippet is also equivalent to:
```python
import graphene

class Character(graphene.ObjectType):
    name = graphene.String(required=True)
```

---
The second one keeps our code consistent since we're using it everywhere and there's no real advantage on using `NonNull` over `required`.

Reply:
and fixed, you can have a look.